opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -61,15 +62,13 @@ namespace Opm
             allcells_[cell] = cell;
         }
 #if HAVE_MPI
-        if ( terminal_output_ ) {
-            if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
-            {
-                const ParallelISTLInformation& info =
-                    boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
-                // Only rank 0 does print to std::cout
-                terminal_output_ = ( info.communicator().rank() == 0 );
-                is_parallel_run_ = ( info.communicator().size() > 1 );
-            }
+        if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
+        {
+            const ParallelISTLInformation& info =
+                boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
+            // Only rank 0 does print to std::cout
+            terminal_output_ &= ( info.communicator().rank() == 0 );
```

Comment:
You can't use `&=` (bitwise operator) here.  You need to say
```C++
terminal_output_ = terminal_output_ && (/* other condition */);
```
It's a shame that there is no corresponding update operator for `bool`s.